### ISSUE TITLE
[Enhancement] Change command sender info with /execute

### ISSUE THREAD

**Issue by DerpNerb** [JorelAli/1.13-Command-API#8]
With the /execute as <entity selector> run <command> you can change the command sender in vanilla. This doesn't work in this API.

Data of the command sender should change with:

- execute at <entity> (changes the command senders location, not the sender itself)
- execute positioned <xyz> (changes the command senders location, not the sender itself)
- execute positioned as <entity> (changes the sender, not the location of it)
- execute rotated <pitch,yaw> (changes sender rotation)
- execute rotated as <entity> (changes sender rotation)
- execute anchored <eyes/feet> (changes sender location)
- execute in <overworld,the_nether,the_end> (changes sender world)
- execute facing <xyz> (changes sender rotation)
- execute facing <entity> (changes sender rotation)

More about the execute command [here](https://minecraft.gamepedia.com/Commands/execute)

**Examples:**

- execute as @e[type=chicken] run say hi (will let all chickens say hi)
- execute as @e[type=chicken] at @s run tp ~ ~1 ~ (will tp all chickens 1 block up)
- execute positioned ~ ~1 ~ run tp ~ ~ ~ (changes sender location to 1 up, so the tp command will tp the sender 1 block up)
- execute in the_nether run tp ~ ~~  (will tp the player to the same location in the nether)

The fact that this doesn't work is such a bummer, because it's the main reason why I want to use this API, **because it can be used in combination with vanilla commands!**

This will be one of the main reasons why people would want to use the API!

Also, adding this feature wouldn't change anything about the normal developer (I think the normal developer doesn't really use vanilla commands), because it only changes the information of the sender when the /execute command is used. 
So just using /<command> wouldn't change anything!
(So I think there's no reason not to do it!)

Can you please change this?

**Comment by JorelAli** [comment]
Thanks for the suggestion. I've begun work on researching this.
It's not that I didn't implement this because I didn't want to, it's rather the fact that I didn't know about the /execute command and have to research how it handles these extra data attributes.

You can keep up to date on the latest developments in [this branch](https://github.com/JorelAli/1.13-Command-API/tree/execute), but I'll try and keep you posted of any major breakthroughs here.

**Comment by Combustible** [comment]
I originally came here looking for this same thing.

Thing is - this API absolutely works with all of the execute variants. I'm using them, and it works great. However - you must modify your target command to accept an EntitySelectorArgument. If you do - then it works! Well, I've tested at least `as`, `at`, `positioned as`. I suspect others work too. 

The thing I originally wanted (and perhaps what the reporter is expecting) is to be able to reach into the "sender" object and get all of this information (because you used to be able to with Spigot's API and ProxiedCommandSender). I had to modify my commands to accept an  EntitySelectorArgument to make things work, which was a pain but I am very happy with it. 

For example I used to have: 
```/execute @e[type=chicken] ~ ~ ~ bossfight mycoolboss ~ ~5 ~ ```
Where the last ~ ~5 ~ were relative to the chicken. Note my command only used to work with /execute

Now all of these work: 
```/bossfight @e[type=chicken] mycoolboss ~ ~5 ~```
Coordinates relative to sender, not chicken

```/execute as @e[type=chicken] run mycoolboss @s ~ ~5 ~```
Works same as above

```/execute as @e[type=chicken] at @s run mycoolboss @s ~ ~5 ~```
Coordinates now relative to the chicken

This IMO feels exactly like how the new vanilla commands work - and so I am happy with it. Where my command used to feel really odd before compared to vanilla commands it now fits right in. 

TLDR: I don't think any changes are needed besides possibly bugfixes. 




However, the more I play with this system

**Comment by Combustible** [comment]
In fact - even though I came here with this in mind, I think this is an anti-feature. Because if you override the "sender" object with the target of execute, you lose information that you otherwise might have used (like the ability to send the command block running a command debug info). 

Take this command I used to have in 1.12:
```/execute bob ~ ~ ~ debuginfo```
This used to send me (the sender!) debug information about the target. Now both of these work after I upgraded: 
```
/debuginfo bob
/execute as bob run debuginfo @s
```
 
If we mangle the sender object to point to the target of execute, the latter case will no longer work. Which seems like a regression to me.

**Comment by HielkePlugins** [comment]
Ah so it is possible with the EntitySelectorArgument? That's cool! 
Still it would be better if it just worked without using that, because that's the way vanilla commands work as well and everyone would expect it to work that way with this API.

And I don't think it's an anti-feature. If we want custom commands to run in combination with vanilla ones, the custom commands should behave like vanilla commands. **Right now this is not the case, and this only causes confusion.**

And in this case:
`/execute as bob run debuginfo @s`

It's entirely logical that bob will get the debuginfo, because **he is the entity the command executed from**! Not the initial sender! If you want the debuginfo from bob, the execute isn't the right command to use. You should just use:
`/debuginfo bob`

You shouldn't run /execute commands if you don't want to change the sender.

The feature definitely isn't a anti-feature!

**Comment by byron-marohn** [comment]
Ok, I'm definitely happy with that solution (using a ProxiedCommandSender). This provides the flexibility to be able to target either the sender or execute target as the developer needs. And like the way spigot worked in the past, ProxiedCommandSender.sendMessage() would by default send to the caller, not callee?

**Comment by Combustible** [comment]
Ok, I'm definitely happy with that solution (using a ProxiedCommandSender). This provides the flexibility to be able to target either the sender or execute target as the developer needs. And like the way spigot worked in the past, ProxiedCommandSender.sendMessage() would by default send to the caller, not callee?

**Comment by JorelAli** [comment]
As of [this commit](https://github.com/JorelAli/1.13-Command-API/commit/793e9945bd4a87c60ca77f379dfccd8738f57a04), I've fixed ProxiedCommandSender's from being generated incorrectly.

_From version 1.3 onwards, when using `/execute as`, it will generate a ProxiedCommandSender_
